### PR TITLE
Moving GCM and InfluxDB to the new external sinks API.

### PR DESCRIPTION
**Pull request by vishh**
Updated integration tests to use the new metrics.

### CODE REVIEW

**Comment by vishh** [comment]
Ping @vmarmol @rjnagal

**Comment by vishh** [comment]
@vmarmol: How can I test GCM?

File: sinks/influxdb/driver.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	writeFailures int // guarded by stateLock
+	avoidColumns  bool
+}
+
+func (self *influxdbSink) Register(metrics []sink_api.MetricDescriptor) error {
+	// Create tags once influxDB v0.9.0 is released.
+	return nil
+}
+
+func (self *influxdbSink) labelsToString(labels map[string]string) string {
```
Comment (vmarmol) [inline sinks/influxdb/driver.go]: Don't we have another version of this in API? Lets make it public and use it here. Or place it in util.

File: sinks/gcm/driver.go [hunk 2]
```diff
@@ -384,3 +256,42 @@ func sendRequest(url string, token string, request interface{}) error {
...
 }
+
+func (self *gcmSink) DebugInfo() string {
+	return "Sink Type: GCM"
+}
+
+// Returns a thread-compatible implementation of GCM interactions.
+func NewSink() (sink_api.ExternalSink, error) {
+	// TODO: Document why this sleep is necessary.
+	time.Sleep(3 * time.Second)
```
Comment (vmarmol) [inline sinks/gcm/driver.go]: LOL this slipped in, I saw flaked with OnGCE() so we should probably try it for ~15s before quitting. TODO is fine

File: sinks/external.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+			errors = append(errors, err)
+		}
+	}
+	err = nil
+	if len(errors) > 0 {
+		errStr := ""
+		for _, err := range errors {
+			errStr = fmt.Sprintf("%v ", err)
+		}
+		err = fmt.Errorf("encountered following errors: %s", errStr)
```
Comment (vmarmol) [inline sinks/external.go]: nit: encountered "the" following

File: sinks/external.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	argDbName       = flag.String("sink_influxdb_name", "k8s", "Influxdb database name")
+	argAvoidColumns = flag.Bool("sink_influxdb_no_columns", false, "When true, prefixes metric series names with metadata instead of storing metadata in additional columns. Metadata includes hostname, container name, etc. ")
+)
+
+type externalSinkManager struct {
+	decoder       sink_api.Decoder
+	externalSinks []sink_api.ExternalSink
+}
+
+func newExternalSink(externalSinks []sink_api.ExternalSink) (ExternalSinkManager, error) {
```
Comment (vmarmol) [inline sinks/external.go]: nit: newExternalSinkManager?

**Comment by vmarmol** [comment]
Minor nits and {subtle} this could have been 2-3 smaller PRs {/subtle} :)

File: sinks/influxdb/driver.go (hunk 1 above)
Comment (vishh) [inline sinks/influxdb/driver.go]: Done

File: sinks/gcm/driver.go (hunk 2 above)
Comment (vishh) [inline sinks/gcm/driver.go]: Added a TODO :)

File: sinks/external.go (hunk 3 above)
Comment (vishh) [inline sinks/external.go]: Done

File: sinks/external.go (hunk 4 above)
Comment (vishh) [inline sinks/external.go]: Done

**Comment by vishh** [comment]
Thanks for the review @vmarmol. I ended up combining to reduce testing time. Apologies :)

**Comment by vmarmol** [comment]
LGTM. Haha no worries, didn't want to let you off without a passive-aggressive comment ;) It was shorter than expected.

**Pull request merged**